tests/e2e/support/objects/app-files/resource/actions.ts

```diff
@@ -447,3 +515,52 @@ export const showHiddenResources = async (page): Promise<void> => {
   await page.locator(filesViewOptionButton).click()
   await page.locator(hiddenFilesToggleButton).click()
 }
+
+export interface editResourcesArgs {
+  page: Page
+  name: string
+  content: string
+}
+
+export const editResources = async (args: editResourcesArgs): Promise<void> => {
+  const { page, name, content } = args
+  await page.locator(util.format(resourceNameSelector, name)).click()
+  await editTextDocument({ page, content: content })
+}
+
+export interface openFileInViewerArgs {
+  page: Page
+  name: string
+  actionType: 'mediaviewer' | 'pdfviewer'
+}
+
+export const openFileInViewer = async (args: openFileInViewerArgs): Promise<void> => {
+  const { page, name, actionType } = args
+
+  if (actionType === 'mediaviewer') {
+    await Promise.all([
+      page.waitForResponse(
+        (resp) =>
+          resp.url().includes('preview') &&
+          resp.status() === 200 &&
+          resp.request().method() === 'GET'
+      ),
+      page.locator(util.format(resourceNameSelector, name)).click()
+    ])
+
+    const previewImage = page.locator(
+      `//main[@id="preview"]//div[contains(@class,"preview-player")]//img`
```

Comment:
maybe we can add this above as a constant

Reply:
done